packages/teleterm/src/ui/QuickInput/useQuickInput.ts

```diff
@@ -16,34 +16,64 @@ limitations under the License.
 
 import React, { useEffect } from 'react';
 
+import { useAsync } from 'shared/hooks/useAsync';
+
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 import {
   useKeyboardShortcuts,
   useKeyboardShortcutFormatters,
 } from 'teleterm/ui/services/keyboardShortcuts';
 import {
-  AutocompleteResult,
-  AutocompletePartialMatch,
+  AutocompleteCommand,
+  AutocompleteToken,
+  Suggestion,
 } from 'teleterm/ui/services/quickInput/types';
 import { routing } from 'teleterm/ui/uri';
 import { KeyboardShortcutType } from 'teleterm/services/config';
 
+import { retryWithRelogin } from '../utils';
+
 export default function useQuickInput() {
-  const { quickInputService, workspacesService, commandLauncher } =
-    useAppContext();
+  const appContext = useAppContext();
+  const { quickInputService, workspacesService, commandLauncher } = appContext;
   workspacesService.useState();
   const documentsService =
     workspacesService.getActiveWorkspaceDocumentService();
   const { visible, inputValue } = quickInputService.useState();
   const [activeSuggestion, setActiveSuggestion] = React.useState(0);
-  const autocompleteResult = React.useMemo(
-    () => quickInputService.getAutocompleteResult(inputValue),
+
+  const parseResult = React.useMemo(
+    () => quickInputService.parse(inputValue),
     // `localClusterUri` has been added to refresh suggestions from
     // `QuickSshLoginPicker` and `QuickServerPicker` when it changes
     [inputValue, workspacesService.getActiveWorkspace()?.localClusterUri]
   );
+
+  const [suggestionsAttempt, getSuggestions] = useAsync(() =>
+    retryWithRelogin(
+      appContext,
+      workspacesService.getActiveWorkspace()?.localClusterUri,
+      () => parseResult.getSuggestions()
+    )
+  );
+
+  useEffect(() => {
+    async function get() {
+      const [, err] = await getSuggestions();
+      if (err && err.name !== 'CanceledError') {
+        console.log('err', err);
```

Comment:
```suggestion
        console.error(err);
```

Reply:
I think we can just remove the console.log anyway. It was left over from development.